### PR TITLE
Fix upgradable storage layout, add etherscan endpoints for L2s

### PR DESCRIPTION
**Pull request by sanderpick**
See comments

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -19,5 +19,7 @@ OPTIMISM_KOVAN_STAGING_API_KEY=fixme
 REPORT_GAS=false
 # contract verification
 ETHERSCAN_API_KEY=fixme
+OPTIMISM_ETHERSCAN_API_KEY=fixme
+POLYSCAN_API_KEY=fixme
```
Comment (sanderpick) [inline .env.example:23]: @joewagner i added these... they are in 1pass

Comment (sanderpick) [reply .env.example:23]: that's all that's needed, you can run the verify script as normal now

File: contracts/TablelandTables.sol [hunk 2]
```diff
@@ -12,26 +12,30 @@ import "./ITablelandTables.sol";
...
+ * @dev V1 storage layout.
  */
-contract TablelandTables is
+abstract contract TablelandTablesV1Storage is
     ITablelandTables,
     ERC721AUpgradeable,
     ERC721AQueryableUpgradeable,
     OwnableUpgradeable,
     PausableUpgradeable,
-    ReentrancyGuardUpgradeable,
```
Comment (sanderpick) [inline contracts/TablelandTables.sol:23]: see https://forum.openzeppelin.com/t/new-storage-layout-is-incompatible-contextupgradeable-sol-inserted-gap/25371

tl;dr, you can't extend here because this contract adds storage, which conflicts with our own slots. i copied the workaround from the issue above. ugly, but we can remove this when we go to mainnet. ideally, we use the diamond storage layout pattern which doesn't have the problem (as easily).

File: package.json [hunk 3]
```diff
@@ -9,13 +9,12 @@
   "types": "typechain-types/index.d.ts",
   "files": [
     "contracts",
-    "!contracts/test",
```
Comment (sanderpick) [inline package.json:12]: may as well include these as examples

File: package.json [hunk 4]
```diff
@@ -9,13 +9,12 @@
...
     "typechain-types/**/*.js?(.map)",
     "typechain-types/**/*.d.ts"
   ],
   "scripts": {
     "build": "npx tsc -p ./tsconfig.build.json",
-    "clean": "rm -rf artifacts && rm -rf typechain-types && rm -rf cache && rm -rf coverage && rm coverage.json",
+    "clean": "rm -rf artifacts && rm -rf typechain-types && rm -rf cache && rm -rf coverage && rm -f coverage.json",
```
Comment (joewagner) [inline package.json:17]: Nice!

**Review by joewagner — CHANGES_REQUESTED** [review]
Looks like prettier and some tests are failing

**Comment by sanderpick** [comment]
@joewagner fixed issues

**Comment by sanderpick** [comment]
@joewagner i pushed the change to publish the hardhat config. Can you give this a quick thumbs up if it looks good to you?

**Review by joewagner — APPROVED** [review]
This all looks to me
